scripts/build_renewable_profiles.py

```diff
@@ -263,21 +267,77 @@
     if snakemake.wildcards.technology.startswith("hydro"):
         country_shapes = gpd.read_file(paths.country_shapes)
         hydrobasins = gpd.read_file(resource["hydrobasins"])
+        ppls = load_powerplants(snakemake.input.powerplants)
+
+        hydro_ppls = ppls[ppls.carrier == "hydro"]
+
         # commented out as rivers may span across multiple countries
         # hydrobasins = hydrobasins[
         #     [
         #         any(country_shapes.geometry.intersects(geom))
         #         for geom in hydrobasins.geometry
         #     ]
         # ]  # exclude hydrobasins shapes that do not intersect the countries of interest
-        resource["plants"] = regions.rename(columns={"x": "lon", "y": "lat"}).loc[
+
+        resource["plants"] = regions.rename(
+            columns={"x": "lon", "y": "lat", "country": "countries"}
+        ).loc[
             [
                 # select busbar whose location (p) belongs to at least one hydrobasin geometry
-                any(hydrobasins.geometry.intersects(p))
-                for p in gpd.points_from_xy(regions.x, regions.y, crs=regions.crs)
+                # if extendable option is true, all buses are included
+                # otherwise only where hydro powerplants are available are considered
+                (
+                    (
+                        config.get("extendable", False)
+                        | (bus_id in hydro_ppls.bus.values)
+                    )
+                    & any(hydrobasins.geometry.intersects(p))
+                )
+                for (p, bus_id) in zip(
+                    gpd.points_from_xy(regions.x, regions.y, crs=regions.crs),
+                    regions.index,
+                )
             ],
-            ["lon", "lat"],
+            ["lon", "lat", "countries"],
         ]  # TODO: filtering by presence of hydro generators should be the way to go
+        resource["plants"]["installed_hydro"] = [
+            True if (bus_id in hydro_ppls.bus.values) else False
+            for bus_id in resource["plants"].index
+        ]
+
+        # check if normalization field belongs to the settings and it is not false
+        if ("normalization" in resource) & (resource["normalization"] is not bool):
+
+            normalization = resource.pop("normalization")
+
+            if normalization == "hydro_capacities":
+                hydro_stats = (
+                    pd.read_csv(
+                        snakemake.input.hydro_capacities,
```

Comment:
would it make sense to read in this snakemake input just below the __main__?

Reply:
We can do that, though I check the possible readability when generalizing this.
Currently I'm on a train with poor connection, hard to work on this

Reply:
As long as it is main, I think we could keep using snakemake.{smt} as long as that parameter is used only once or twice in the code.
I think that initializing variables at the beginning that are used only once may create only large chuncks of variables at the beginning with scarce benefits, what do you think?